Fix GradBucket Typing

mrshenli opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#52943 Fix GradBucket Typing**

Differential Revision: [D26699759](https://our.internmc.facebook.com/intern/diff/D26699759)

mrshenli committed:
Fix GradBucket Typing

[ghstack-poisoned]

Commit:
Update _distributed_c10d.pyi

Adjusted the order of these declaration to be consistent with the order in c10d/init.cpp.

SciPioneer commented on torch/distributed/algorithms/ddp_comm_hooks/powerSGD_hook.py:
```diff
@@ -162,7 +162,9 @@ def maybe_increase_iter(self, bucket):
             )
 
 
-def powerSGD_hook(state: PowerSGDState, bucket) -> torch.futures.Future:
```
Can you also add the typing for another hook batched_powerSGD_hook in the same file?

wayi1 reviewed (approved):
Thanks for the fix!

mrshenli committed:
Update on "Fix GradBucket Typing"


Differential Revision: [D26699759](https://our.internmc.facebook.com/intern/diff/D26699759)

[ghstack-poisoned]
